File: proyecto-backend-nexu/src/main/java/com/nexu/PlacesToFun/dto/PlacesToFunRequestDto.java
package com.nexu.PlacesToFun.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.nexu.Places.domain.enums.Payment;
import com.nexu.PlacesToFun.domain.enums.PlaceCategoryToFun;
import com.nexu.PlacesToFun.domain.enums.SizePark;
import jakarta.validation.constraints.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class PlacesToFunRequestDto {

    @NotBlank(message = "El nombre es obligatorio")
    private String name;

    @NotBlank(message = "La dirección es obligatoria")
    private String address;

    @NotNull
    private Double latitude;

    @NotNull
    private Double longitude;

    @NotNull
    private Payment payment;

    @NotNull
    @JsonFormat(pattern = "HH:mm")
    private LocalTime openTime;

    @NotNull
    @JsonFormat(pattern = "HH:mm")
    private LocalTime closeTime;

    @NotBlank
    private String description;

    private Boolean wifi;

    private String priceRange;

    private Double estimatedPrice;

    @NotNull
    private Integer capacity;

    @NotBlank
    private String status;

    // específicos de PlacesToFun
    @NotNull
    private PlaceCategoryToFun placeCategoryToFun;
    private List<String> games;
    private Double priceFicha;
    private SizePark sizePark;
    private Boolean haveGames;
}

File: proyecto-frontend-mobile-nexu3/src/interfaces/placesToFun/PlacesToFunRequestDto.ts
export interface PlacesToFunRequestDto {
  name: string;
  address: string;
  latitude: number;
  longitude: number;
  payment: Payment;
  openTime: string; // HH:mm
  closeTime: string; // HH:mm
  description: string;
  wifi?: boolean;
  priceRange?: string;
  estimatedPrice?: number;
  capacity: number;
  status: string;
  placeCategoryToFun: PlaceCategoryToFun;
  games?: string[];
  priceFicha?: number;
  sizePark?: SizePark;
  haveGames?: boolean;
}

export type Payment = 'FREE' | 'CARD' | 'CASH' | 'YAPE';
export type PlaceCategoryToFun = 'PARK' | 'GAMES';
export type SizePark = 'BIG' | 'REGULAR' | 'SMALL'; 